graph/repo.go

```diff
@@ -33,6 +33,14 @@ func TryMakeURI(cloneURL string) (string, error) {
 	url, err := url.Parse(cloneURL)
 	if err != nil {
 		return "", fmt.Errorf("MakeURI(%q): %s", cloneURL, err)
+	} else if url.Path == "" || url.Path == "/" {
+		return "", fmt.Errorf("MakeURI(%q): missing path from URL", cloneURL)
+	} else if url.Host == "" && !strings.Contains(strings.Split(url.Path, "/")[0], ".") {
```

Comment:
This assumes that the hostname has a dot, which is not true in the general case (e.g., for localhost, or internal server hostnames). While it's unlikely that someone would want a repo URI like `localhost/foo/bar`, it's reasonable that they'd have `http://gitrepos` be an intranet host and thus would have `gitrepos/foo/bar` be a repo URI. 

I think it's fine to adopt the following lesser constraint: the host + "/" + path must not be empty. If it is empty, then return an error.